modules/codegen-plugin/src/smithy4s/codegen/Smithy4sCodegenPlugin.scala

```diff
@@ -134,9 +199,78 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
       artifactFile <- module.artifacts
     } yield {
       val (_, file) = artifactFile
-      os.Path(file)
+      file
+    }
+
+  private def moduleIdEncode(
+      moduleId: ModuleID,
+      scalaBinaryVersion: Option[String]
+  ): List[String] = {
+    (moduleId.crossVersion, scalaBinaryVersion) match {
+      case (Disabled, _) =>
+        List(s"${moduleId.organization}:${moduleId.name}:${moduleId.revision}")
+      case (_: Binary, Some(sbv)) =>
+        List(
+          s"${moduleId.organization}:${moduleId.name}_${sbv}:${moduleId.revision}"
+        )
+      case (_, _) => Nil
+    }
+  }
+
+  /**
+   * Retrieves the smithy4sDependencies that compile-dependencies may have listed
+   * in their jar manifests when they were packaged.
+   */
+  private def smithy4sFetchUpstreamCodegenDependencies(
+      config: Configuration
+  ): Def.Initialize[Task[Seq[File]]] =
+    Def.task {
+      val smithy4sDependencies =
+        (config / externalDependencyClasspath).value
+          .map(_.data)
+          .flatMap(extract)
+      def getJars(ids: Seq[ModuleID]): Seq[File] = {
+        val syntheticModule =
+          organization.value % (name.value + "-smithy4s-resolution") % version.value
+        val depRes = (update / dependencyResolution).value
+        val updc = (update / updateConfiguration).value
+        val uwconfig = (update / unresolvedWarningConfiguration).value
+        val modDescr = depRes.moduleDescriptor(
+          syntheticModule,
+          ids.toVector,
+          None
+        )
+
+        depRes
+          .update(
+            modDescr,
+            updc,
+            uwconfig,
+            streams.value.log
+          )
+          .map(_.allFiles)
+          .fold(uw => throw uw.resolveException, identity)
+      }
+      getJars(smithy4sDependencies)
     }
 
+  private lazy val simple = raw"([^:]*):([^:]*):([^:]*)".r
+  private lazy val cross = raw"([^:]*)::([^:]*):([^:]*)".r
+  private def extract(jarFile: java.io.File): Seq[ModuleID] = {
+    val jar = new JarFile(jarFile)
+    Option(
+      jar.getManifest().getMainAttributes().getValue("smithy4sDependencies")
```

Comment:
let's extract `smithy4sDependencies` to a constant

Reply:
something something draft